apps/voting/contracts/Voting.sol

```diff
@@ -152,10 +176,46 @@ contract Voting is IForwarder, AragonApp {
     * @param _executesIfDecided Whether the vote should execute its action if it becomes decided
     */
     function vote(uint256 _voteId, bool _supports, bool _executesIfDecided) external voteExists(_voteId) {
-        require(_canVote(_voteId, msg.sender), ERROR_CAN_NOT_VOTE);
+        require(_canVote(votes[_voteId], msg.sender), ERROR_CAN_NOT_VOTE);
         _vote(_voteId, _supports, msg.sender, _executesIfDecided);
     }
 
+    /**
+    * @notice Vote `_supports ? 'yes' : 'no'` in vote #`_voteId` on behalf of `_voter`
+    * @dev Initialization check is implicitly provided by `voteExists()` as new votes can only be
+    *      created via `newVote(),` which requires initialization
+    * @param _voter Address of the voter voting on behalf of
+    * @param _voteId Id for vote
+    * @param _supports Whether the representative supports the vote
+    */
+    function voteOnBehalfOf(address _voter, uint256 _voteId, bool _supports) external voteExists(_voteId) {
+        Vote storage vote_ = votes[_voteId];
+        require(_canVoteOnBehalfOf(vote_, _voter, msg.sender), ERROR_REPRESENTATIVE_CANT_VOTE);
+        _voteOnBehalfOf(_voteId, _supports, _voter, msg.sender);
+    }
+
+    /**
+    * @notice Vote `_supports ? 'yes' : 'no'` in vote #`_voteId` on behalf of many voters
+    * @dev Initialization check is implicitly provided by `voteExists()` as new votes can only be
+    *      created via `newVote(),` which requires initialization
+    * @param _voters Addresses of the voters voting on behalf of
+    * @param _voteId Id for vote
+    * @param _supports Whether the representative supports the vote
+    */
+    function voteOnBehalfOfMany(address[] _voters, uint256 _voteId, bool _supports) external voteExists(_voteId) {
```

Comment:
I see what you mean, I see two issues here:
- The representative may not be allowed to vote himself, it could be any address.
- If the representative is allowed, enforcing them through this entry point won't prevent them from changing their vote with the current `vote()` entry point afterwards. And adding a guard there would increase gas costs for all the voters, what do you think?

Reply:
Yes, you are absolutely right.